File: csharp/DVSClient/DVSClient/address/layout/elements/Aug.cs
using System.Reflection;

namespace DVSClient.Address.Layout.Elements
{
    public class Aug : IAddressElement
    {
        public static readonly Aug BuildingName = new Aug("buildingName", "Building name", "Treasury Building");
        public static readonly Aug FlatUnitName = new Aug("flatUnitName", "Flat/Unit name", "Flat 2");
        public static readonly Aug FlatUnitType = new Aug("flatUnitType", "Flat/Unit type", "Flat");
        public static readonly Aug FlatUnitNumber = new Aug("flatUnitNumber", "Flat/unit number", "2");
        public static readonly Aug SubBuildingNumber = new Aug("subBuildingNumber", "Sub-building number", "5a");
        public static readonly Aug SubBuildingNumberNumber = new Aug("subBuildingNumberNumber", "Sub-building number (number)", "5");
        public static readonly Aug SubBuildingNumberAlpha = new Aug("subBuildingNumberAlpha", "Sub-building number (alpha)", "a");
        public static readonly Aug BuildingLevel = new Aug("buildingLevel", "Building level", "Level 7");
        public static readonly Aug BuildingLevelType = new Aug("buildingLevelType", "Building level type", "Level");
        public static readonly Aug BuildingLevelNumber = new Aug("buildingLevelNumber", "Building level number", "7");
        public static readonly Aug BuildingNumber = new Aug("buildingNumber", "Building number", "1-131");
        public static readonly Aug BuildingNumberFirst = new Aug("buildingNumberFirst", "Building number (first)", "1");
        public static readonly Aug BuildingNumberLast = new Aug("buildingNumberLast", "Building number (last)", "131");
        public static readonly Aug AllotmentNumber1 = new Aug("allotmentNumber1", "Allotment number", "Lot 16");
        public static readonly Aug AllotmentLot = new Aug("allotmentLot", "Allotment lot", "Lot");
        public static readonly Aug AllotmentNumber2 = new Aug("allotmentNumber2", "Allotment number", "16");
        public static readonly Aug Street = new Aug("street", "Street", "Tudor Court East");
        public static readonly Aug StreetName = new Aug("streetName", "Street name", "Tudor");
        public static readonly Aug StreetType = new Aug("streetType", "Street type", "Court");
        public static readonly Aug StreetTypeSuffix = new Aug("streetTypeSuffix", "Street type suffix", "East");
        public static readonly Aug PrivateStreet = new Aug("privateStreet", "Private Street", "Private Street");
        public static readonly Aug Locality = new Aug("locality", "Locality", "Ayr");
        public static readonly Aug BorderingLocality = new Aug("borderingLocality", "Bordering Locality", "Mt Kelly");
        public static readonly Aug StateCode = new Aug("stateCode", "State code", "QLD");
        public static readonly Aug StateName = new Aug("stateName", "State Name", "Queensland");
        public static readonly Aug Postcode = new Aug("postcode", "Postcode", "4807");
        public static readonly Aug CountryName = new Aug("countryName", "Country Name", "Australia");
        public static readonly Aug TwoCharacterIsoCountryCode = new Aug("twoCharacterIsoCountryCode", "Two character country code", "AU");
        public static readonly Aug ThreeCharacterIsoCountryCode = new Aug("threeCharacterIsoCountryCode", "Three character country code", "AUS");
        public static readonly Aug GeocodeLevelCode = new Aug("geocodeLevelCode", "This is the geocode level code", "2");
        public static readonly Aug GeocodeLevelDescription = new Aug("geocodeLevelDescription", "This is the geocode level description", "Street level geocode only");
        public static readonly Aug GeocodeTypeCode = new Aug("geocodeTypeCode", "This is the geocode type code", "LB");
        public static readonly Aug GeocodeTypeDescription = new Aug("geocodeTypeDescription", "This is the geocode type description", "Letterbox");
        public static readonly Aug AddressLevelLongitude = new Aug("addressLevelLongitude", "The address-level longitude in degrees", "");
        public static readonly Aug AddressLevelLatitude = new Aug("addressLevelLatitude", "The address-level latitude in degrees", "");
        public static readonly Aug AddressLevelElevation = new Aug("addressLevelElevation", "The address-level elevation", "");
        public static readonly Aug AddressLevelPlanimetricAccuracy = new Aug("addressLevelPlanimetricAccuracy", "The address-level planimetric accuracy", "");
        public static readonly Aug AddressLevelBoundaryExtent = new Aug("addressLevelBoundaryExtent", "The address-level boundary extent", "");
        public static readonly Aug AddressLevelGeocodeReliabilityCode = new Aug("addressLevelGeocodeReliabilityCode", "The address-level geocode reliability code", "2");
        public static readonly Aug AddressLevelGeocodeReliabilityDescription = new Aug("addressLevelGeocodeReliabilityDescription", "The address-level geocode reliability description", "Geocode accuracy sufficient to place centroid within address site boundary");
        public static readonly Aug StreetLevelLongitude = new Aug("streetLevelLongitude", "The street-level longitude in degrees", "");
        public static readonly Aug StreetLevelLatitude = new Aug("streetLevelLatitude", "The street-level latitude in degrees", "");
        public static readonly Aug StreetLevelPlanimetricAccuracy = new Aug("streetLevelPlanimetricAccuracy", "The street-level planimetric accuracy", "");
        public static readonly Aug StreetLevelBoundaryExtent = new Aug("streetLevelBoundaryExtent", "The street-level boundary extent", "");
        public static readonly Aug StreetLevelGeocodeReliabilityCode = new Aug("streetLevelGeocodeReliabilityCode", "The street-level geocode reliability code", "4");
        public static readonly Aug StreetLevelGeocodeReliabilityDescription = new Aug("streetLevelGeocodeReliabilityDescription", "The street-level geocode reliability description", "Geocode accuracy sufficient to associate address site with a unique road feature");
        public static readonly Aug LocalityLevelLongitude = new Aug("localityLevelLongitude", "The locality-level longitude in degrees", "");
        public static readonly Aug LocalityLevelLatitude = new Aug("localityLevelLatitude", "The locality-level latitude in degrees", "");
        public static readonly Aug LocalityLevelPlanimetricAccuracy = new Aug("localityLevelPlanimetricAccuracy", "The locality-level planimetric accuracy", "");
        public static readonly Aug LocalityLevelGeocodeReliabilityCode = new Aug("localityLevelGeocodeReliabilityCode", "The locality-level geocode reliability code", "5");
        public static readonly Aug LocalityLevelGeocodeReliabilityDescription = new Aug("localityLevelGeocodeReliabilityDescription", "The locality-level geocode reliability description", "Geocode accuracy sufficient to associate address site with a unique locality or neighbourhood");
        public static readonly Aug Longitude = new Aug("longitude", "The highest-level longitude in degrees", "");
        public static readonly Aug Latitude = new Aug("latitude", "The highest-level latitude in degrees", "");
        public static readonly Aug Elevation = new Aug("elevation", "The highest-level elevation", "");
        public static readonly Aug PlanimetricAccuracy = new Aug("planimetricAccuracy", "The highest-level planimetric accuracy", "");
        public static readonly Aug BoundaryExtent = new Aug("boundaryExtent", "The highest-level boundary extent", "");
        public static readonly Aug GeocodeReliabilityCode = new Aug("geocodeReliabilityCode", "The highest-level geocode reliability code", "");
        public static readonly Aug GeocodeReliabilityDescription = new Aug("geocodeReliabilityDescription", "The highest-level geocode reliability description", "");
        public static readonly Aug GnafPid = new Aug("gnafPid", "Persistent identifier of an address", "GANSW716798454");
        public static readonly Aug GnafAddressTypeCode = new Aug("gnafAddressTypeCode", "This is the address type code", "R/RMB");
        public static readonly Aug GnafAddressTypeDescription = new Aug("gnafAddressTypeDescription", "This is the address type description", "Rural Roadside Mail Box");
        public static readonly Aug StreetPid = new Aug("streetPid", "This is a unique street persistent identifier", "");
        public static readonly Aug LocalityPid = new Aug("localityPid", "This is a unique locality persistent identifier", "");
        public static readonly Aug ConfidenceLevelCode = new Aug("confidenceLevelCode", "This is the confidence level code", "2");
        public static readonly Aug ConfidenceLevelDescription = new Aug("confidenceLevelDescription", "This is the confidence level descriptor", "All three contributors have supplied an identical address");
        public static readonly Aug MeshBlockId2021 = new Aug("2021MeshBlockId", "The 2021 version of the Mesh Block ID", "");
        public static readonly Aug MeshBlockCode2021 = new Aug("2021MeshBlockCode", "The 11-digit 2021 version of the Mesh Block Code", "");
        public static readonly Aug MeshBlockMatchCode2021 = new Aug("2021MeshBlockMatchCode", "The code for the level of matching to 2021 Mesh Blocks", "");
        public static readonly Aug MeshBlockMatchDescription2021 = new Aug("2021MeshBlockMatchDescription", "The description of the 2021 Mesh Block match level", "");
        public static readonly Aug MeshBlockId2016 = new Aug("2016MeshBlockId", "The 2016 version of the MeshBlock ID", "");
        public static readonly Aug MeshBlockCode2016 = new Aug("2016MeshBlockCode", "The 11-digit 2016 version of the MeshBlock Code", "");
        public static readonly Aug MeshBlockMatchCode2016 = new Aug("2016MeshBlockMatchCode", "The code for the level of matching to 2016 Mesh Blocks", "");
        public static readonly Aug MeshBlockMatchDescription2016 = new Aug("2016MeshBlockMatchDescription", "The description of the 2016 Mesh Block match level", "");
        public static readonly Aug ComplexAddressTypeCode = new Aug("complexAddressTypeCode", "Returns the Address Type", "P");
        public static readonly Aug PrimaryAddressPid = new Aug("primaryAddressPid", "This element will only be populated if the input address is a secondary address", "");
        public static readonly Aug AddressJoinTypeCode = new Aug("addressJoinTypeCode", "Returns the Join Type Code", "");
        public static readonly Aug GnafLegalParcelIdentifier = new Aug("gnafLegalParcelIdentifier", "The Legal Parcel Id field", "");
        public static readonly Aug CollectorDistrictId = new Aug("collectorDistrictId", "This is a unique Collector District persistent identifier", "");
        public static readonly Aug CollectorDistrictCode = new Aug("collectorDistrictCode", "This is the Collector District Code", "");
        public static readonly Aug CommonwealthElectoralBoundaryId = new Aug("commonwealthElectoralBoundaryId", "This is a unique Commonwealth Electoral Boundary persistent identifier", "");
        public static readonly Aug CommonwealthElectoralBoundaryName = new Aug("commonwealthElectoralBoundaryName", "This is the Commonwealth Electoral Boundary name", "");
        public static readonly Aug LocalGovernmentAreaId = new Aug("localGovernmentAreaId", "This is a Local Government Area persistent identifier", "");
        public static readonly Aug LocalGovernmentAreaName = new Aug("localGovernmentAreaName", "This is the Local Government Area name", "");
        public static readonly Aug StatisticalLocalAreaId = new Aug("statisticalLocalAreaId", "This is a Statistical Local Area persistent identifier", "");
        public static readonly Aug StatisticalLocalAreaCode = new Aug("statisticalLocalAreaCode", "This is a Statistical Local Area code", "");
        public static readonly Aug StatisticalLocalAreaName = new Aug("statisticalLocalAreaName", "This is a Statistical Local Area name", "");
        public static readonly Aug StateElectoralBoundaryId = new Aug("stateElectoralBoundaryId", "This is a State Electoral Boundary persistent identifier", "");
        public static readonly Aug StateElectoralBoundaryName = new Aug("stateElectoralBoundaryName", "This is a State Electoral Boundary name", "");
        public static readonly Aug StateElectoralEffectiveStart = new Aug("stateElectoralEffectiveStart", "This is the date that the electorate becomes effective", "");
        public static readonly Aug StateElectoralEffectiveEnd = new Aug("stateElectoralEffectiveEnd", "This is the end date when electorate is no longer in effect", "");
        public static readonly Aug StateElectoralNewBoundaryId = new Aug("stateElectoralNewBoundaryId", "This is the State Electoral Boundary identifier for new electorate", "");
        public static readonly Aug StateElectoralNewBoundaryName = new Aug("stateElectoralNewBoundaryName", "This is the State Electoral Boundary name for new electorate", "");
        public static readonly Aug StateElectoralNewEffectiveStart = new Aug("stateElectoralNewEffectiveStart", "This is the start date that the new electorate will become effective", "");
        public static readonly Aug StateElectoralNewEffectiveEnd = new Aug("stateElectoralNewEffectiveEnd", "This is the end date when the new electorate will no longer be in effect", "");
        public static readonly Aug MosaicGroup = new Aug("mosaicGroup", "Mosaic Group", "K");
        public static readonly Aug MosaicType = new Aug("mosaicType", "Mosaic Group and Type", "K39");
        public static readonly Aug MosaicSegmentCode = new Aug("mosaicSegmentCode", "Mosaic Segment", "A01_3");
        public static readonly Aug Factor1Score = new Aug("factor1Score", "Factors score", "-24255");
        public static readonly Aug Factor1Percentile = new Aug("factor1Percentile", "Factors percentile", "13");
        public static readonly Aug Factor2Score = new Aug("factor2Score", "Factors score", "-024255");
        public static readonly Aug Factor2Percentile = new Aug("factor2Percentile", "Factors percentile", "13");
        public static readonly Aug Factor3Score = new Aug("factor3Score", "Factors score", "-24255");
        public static readonly Aug Factor3Percentile = new Aug("factor3Percentile", "Factors percentile", "13");
        public static readonly Aug Factor4Score = new Aug("factor4Score", "Factors score", "-24255");
        public static readonly Aug Factor4Percentile = new Aug("factor4Percentile", "Factors percentile", "13");
        public static readonly Aug Factor5Score = new Aug("factor5Score", "Factors score", "-24255");
        public static readonly Aug Factor5Percentile = new Aug("factor5Percentile", "Factors percentile", "13");
        public static readonly Aug LengthOfResidence = new Aug("lengthOfResidence", "The code representing the Length of Residence band", "2");
        public static readonly Aug HeadOfHouseholdAge = new Aug("headOfHouseholdAge", "The code representing the Age band", "11");
        public static readonly Aug PropensityForChildren0To10Years = new Aug("propensityForChildren0To10Years", "Double digit code representing the band", "02");
        public static readonly Aug PropensityForChildren11To18Years = new Aug("propensityForChildren11To18Years", "Double digit code representing the band", "02");
        public static readonly Aug AdultsAtAddress = new Aug("adultsAtAddress", "An estimate of the number of adults at the address", "3");
        public static readonly Aug HouseholdComposition = new Aug("householdComposition", "Single digit code representing the Household Composition band", "2");
        public static readonly Aug Lifestage = new Aug("lifestage", "2-digit code representing the Lifestage band", "09");
        public static readonly Aug HouseholdIncome = new Aug("householdIncome", "Single digit code representing the Household Income band", "5");
        public static readonly Aug Affluence = new Aug("affluence", "Single digit code representing the Affluence band", "3");
        public static readonly Aug RiskInsight = new Aug("riskInsight", "Code representing the Risk Insight band", "7");
        public static readonly Aug CreditDemand = new Aug("creditDemand", "Code representing the Credit Demand band", "7");

        private string elementName { get; set; }
        private string description { get; set; }
        private string example { get; set; }

        public Aug(string elementName, string description, string example)
        {
            this.elementName = elementName;
            this.description = description;
            this.example = example;
        }

        public static IAddressElement? GetElement(string name)
        {
            if (!string.IsNullOrWhiteSpace(name))
            {
                Type addressElementType = typeof(Aug);
                FieldInfo? addressElementProperty = addressElementType.GetField(name);

                if (addressElementProperty != null)
                {
                    return (IAddressElement?)addressElementProperty.GetValue(name);
                }
            }
             
            return null;
        }

        public string GetElementName()
        {
            return elementName;
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/address/layout/elements/Aug.java
package com.experian.dvs.client.address.layout.elements;

public enum Aug implements AddressElement {

    BUILDING_NAME("buildingName", "Building name", "Treasury Building"),
    FLAT_UNIT_NAME("flatUnitName", "Flat/Unit name", "Flat 2"),
    FLAT_UNIT_TYPE("flatUnitType", "Flat/Unit type", "Flat"),
    FLAT_UNIT_NUMBER("flatUnitNumber", "Flat/unit number", "2"),
    SUB_BUILDING_NUMBER("subBuildingNumber", "Sub-building number", "5a"),
    SUB_BUILDING_NUMBER_NUMBER("subBuildingNumberNumber", "Sub-building number (number)", "5"),
    SUB_BUILDING_NUMBER_ALPHA("subBuildingNumberAlpha", "Sub-building number (alpha)", "a"),
    BUILDING_LEVEL("buildingLevel", "Building level", "Level 7"),
    BUILDING_LEVEL_TYPE("buildingLevelType", "Building level type", "Level"),
    BUILDING_LEVEL_NUMBER("buildingLevelNumber", "Building level number", "7"),
    BUILDING_NUMBER("buildingNumber", "Building number", "1-131"),
    BUILDING_NUMBER_FIRST("buildingNumberFirst", "Building number (first)", "1"),
    BUILDING_NUMBER_LAST("buildingNumberLast", "Building number (last)", "131"),
    ALLOTMENT_NUMBER_1("allotmentNumber1", "Allotment number", "Lot 16"),
    ALLOTMENT_LOT("allotmentLot", "Allotment lot", "Lot"),
    ALLOTMENT_NUMBER_2("allotmentNumber2", "Allotment number", "16"),
    STREET("street", "Street", "Tudor Court East"),
    STREET_NAME("streetName", "Street name", "Tudor"),
    STREET_TYPE("streetType", "Street type", "Court"),
    STREET_TYPE_SUFFIX("streetTypeSuffix", "Street type suffix", "East"),
    PRIVATE_STREET("privateStreet", "Private Street", "Private Street"),
    LOCALITY("locality", "Locality", "Ayr"),
    BORDERING_LOCALITY("borderingLocality", "Bordering Locality", "Mt Kelly"),
    STATE_CODE("stateCode", "State code", "QLD"),
    STATE_NAME("stateName", "State Name", "Queensland"),
    POSTCODE("postcode", "Postcode", "4807"),
    COUNTRY_NAME("countryName", "Country Name", "Australia"),
    TWO_CHARACTER_ISO_COUNTRY_CODE("twoCharacterIsoCountryCode", "Two character country code", "AU"),
    THREE_CHARACTER_ISO_COUNTRY_CODE("threeCharacterIsoCountryCode", "Three character country code", "AUS"),
    GEOCODE_LEVEL_CODE("geocodeLevelCode", "This is the geocode level code", "2"),
    GEOCODE_LEVEL_DESCRIPTION("geocodeLevelDescription", "This is the geocode level description", "Street level geocode only"),
    GEOCODE_TYPE_CODE("geocodeTypeCode", "This is the geocode type code", "LB"),
    GEOCODE_TYPE_DESCRIPTION("geocodeTypeDescription", "This is the geocode type description", "Letterbox"),
    ADDRESS_LEVEL_LONGITUDE("addressLevelLongitude", "The address-level longitude in degrees", ""),
    ADDRESS_LEVEL_LATITUDE("addressLevelLatitude", "The address-level latitude in degrees", ""),
    ADDRESS_LEVEL_ELEVATION("addressLevelElevation", "The address-level elevation", ""),
    ADDRESS_LEVEL_PLANIMETRIC_ACCURACY("addressLevelPlanimetricAccuracy", "The address-level planimetric accuracy", ""),
    ADDRESS_LEVEL_BOUNDARY_EXTENT("addressLevelBoundaryExtent", "The address-level boundary extent", ""),
    ADDRESS_LEVEL_GEOCODE_RELIABILITY_CODE("addressLevelGeocodeReliabilityCode", "The address-level geocode reliability code", "2"),
    ADDRESS_LEVEL_GEOCODE_RELIABILITY_DESCRIPTION("addressLevelGeocodeReliabilityDescription", "The address-level geocode reliability description", "GlobalGeocodeAttribute accuracy sufficient to place centroid within address site boundary"),
    STREET_LEVEL_LONGITUDE("streetLevelLongitude", "The street-level longitude in degrees", ""),
    STREET_LEVEL_LATITUDE("streetLevelLatitude", "The street-level latitude in degrees", ""),
    STREET_LEVEL_PLANIMETRIC_ACCURACY("streetLevelPlanimetricAccuracy", "The street-level planimetric accuracy", ""),
    STREET_LEVEL_BOUNDARY_EXTENT("streetLevelBoundaryExtent", "The street-level boundary extent", ""),
    STREET_LEVEL_GEOCODE_RELIABILITY_CODE("streetLevelGeocodeReliabilityCode", "The street-level geocode reliability code", "4"),
    STREET_LEVEL_GEOCODE_RELIABILITY_DESCRIPTION("streetLevelGeocodeReliabilityDescription", "The street-level geocode reliability description", "GlobalGeocodeAttribute accuracy sufficient to associate address site with a unique road feature"),
    LOCALITY_LEVEL_LONGITUDE("localityLevelLongitude", "The locality-level longitude in degrees", ""),
    LOCALITY_LEVEL_LATITUDE("localityLevelLatitude", "The locality-level latitude in degrees", ""),
    LOCALITY_LEVEL_PLANIMETRIC_ACCURACY("localityLevelPlanimetricAccuracy", "The locality-level planimetric accuracy", ""),
    LOCALITY_LEVEL_GEOCODE_RELIABILITY_CODE("localityLevelGeocodeReliabilityCode", "The locality-level geocode reliability code", "5"),
    LOCALITY_LEVEL_GEOCODE_RELIABILITY_DESCRIPTION("localityLevelGeocodeReliabilityDescription", "The locality-level geocode reliability description", "GlobalGeocodeAttribute accuracy sufficient to associate address site with a unique locality or neighbourhood"),
    LONGITUDE("longitude", "The highest-level longitude in degrees", ""),
    LATITUDE("latitude", "The highest-level latitude in degrees", ""),
    ELEVATION("elevation", "The highest-level elevation", ""),
    PLANIMETRIC_ACCURACY("planimetricAccuracy", "The highest-level planimetric accuracy", ""),
    BOUNDARY_EXTENT("boundaryExtent", "The highest-level boundary extent", ""),
    GEOCODE_RELIABILITY_CODE("geocodeReliabilityCode", "The highest-level geocode reliability code", ""),
    GEOCODE_RELIABILITY_DESCRIPTION("geocodeReliabilityDescription", "The highest-level geocode reliability description", ""),
    GNAF_PID("gnafPid", "Persistent identifier of an address", "GANSW716798454"),
    GNAF_ADDRESS_TYPE_CODE("gnafAddressTypeCode", "This is the address type code", "R/RMB"),
    GNAF_ADDRESS_TYPE_DESCRIPTION("gnafAddressTypeDescription", "This is the address type description", "Rural Roadside Mail Box"),
    STREET_PID("streetPid", "This is a unique street persistent identifier", ""),
    LOCALITY_PID("localityPid", "This is a unique locality persistent identifier", ""),
    CONFIDENCE_LEVEL_CODE("confidenceLevelCode", "This is the confidence level code", "2"),
    CONFIDENCE_LEVEL_DESCRIPTION("confidenceLevelDescription", "This is the confidence level descriptor", "All three contributors have supplied an identical address"),
    MESH_BLOCK_ID_2021("2021MeshBlockId", "The 2021 version of the Mesh Block ID", ""),
    MESH_BLOCK_CODE_2021("2021MeshBlockCode", "The 11-digit 2021 version of the Mesh Block Code", ""),
    MESH_BLOCK_MATCH_CODE_2021("2021MeshBlockMatchCode", "The code for the level of matching to 2021 Mesh Blocks", ""),
    MESH_BLOCK_MATCH_DESCRIPTION_2021("2021MeshBlockMatchDescription", "The description of the 2021 Mesh Block match level", ""),
    MESH_BLOCK_ID_2016("2016MeshBlockId", "The 2016 version of the MeshBlock ID", ""),
    MESH_BLOCK_CODE_2016("2016MeshBlockCode", "The 11-digit 2016 version of the MeshBlock Code", ""),
    MESH_BLOCK_MATCH_CODE_2016("2016MeshBlockMatchCode", "The code for the level of matching to 2016 Mesh Blocks", ""),
    MESH_BLOCK_MATCH_DESCRIPTION_2016("2016MeshBlockMatchDescription", "The description of the 2016 Mesh Block match level", ""),
    COMPLEX_ADDRESS_TYPE_CODE("complexAddressTypeCode", "Returns the FormatAddress Type", "P"),
    PRIMARY_ADDRESS_PID("primaryAddressPid", "This element will only be populated if the input address is a secondary address", ""),
    ADDRESS_JOIN_TYPE_CODE("addressJoinTypeCode", "Returns the Join Type Code", ""),
    GNAF_LEGAL_PARCEL_IDENTIFIER("gnafLegalParcelIdentifier", "The Legal Parcel Id field", ""),
    COLLECTOR_DISTRICT_ID("collectorDistrictId", "This is a unique Collector District persistent identifier", ""),
    COLLECTOR_DISTRICT_CODE("collectorDistrictCode", "This is the Collector District Code", ""),
    COMMONWEALTH_ELECTORAL_BOUNDARY_ID("commonwealthElectoralBoundaryId", "This is a unique Commonwealth Electoral Boundary persistent identifier", ""),
    COMMONWEALTH_ELECTORAL_BOUNDARY_NAME("commonwealthElectoralBoundaryName", "This is the Commonwealth Electoral Boundary name", ""),
    LOCAL_GOVERNMENT_AREA_ID("localGovernmentAreaId", "This is a Local Government Area persistent identifier", ""),
    LOCAL_GOVERNMENT_AREA_NAME("localGovernmentAreaName", "This is the Local Government Area name", ""),
    STATISTICAL_LOCAL_AREA_ID("statisticalLocalAreaId", "This is a Statistical Local Area persistent identifier", ""),
    STATISTICAL_LOCAL_AREA_CODE("statisticalLocalAreaCode", "This is a Statistical Local Area code", ""),
    STATISTICAL_LOCAL_AREA_NAME("statisticalLocalAreaName", "This is a Statistical Local Area name", ""),
    STATE_ELECTORAL_BOUNDARY_ID("stateElectoralBoundaryId", "This is a State Electoral Boundary persistent identifier", ""),
    STATE_ELECTORAL_BOUNDARY_NAME("stateElectoralBoundaryName", "This is a State Electoral Boundary name", ""),
    STATE_ELECTORAL_EFFECTIVE_START("stateElectoralEffectiveStart", "This is the date that the electorate becomes effective", ""),
    STATE_ELECTORAL_EFFECTIVE_END("stateElectoralEffectiveEnd", "This is the end date when electorate is no longer in effect", ""),
    STATE_ELECTORAL_NEW_BOUNDARY_ID("stateElectoralNewBoundaryId", "This is the State Electoral Boundary identifier for new electorate", ""),
    STATE_ELECTORAL_NEW_BOUNDARY_NAME("stateElectoralNewBoundaryName", "This is the State Electoral Boundary name for new electorate", ""),
    STATE_ELECTORAL_NEW_EFFECTIVE_START("stateElectoralNewEffectiveStart", "This is the start date that the new electorate will become effective", ""),
    STATE_ELECTORAL_NEW_EFFECTIVE_END("stateElectoralNewEffectiveEnd", "This is the end date when the new electorate will no longer be in effect", ""),
    MOSAIC_GROUP("mosaicGroup", "Mosaic Group", "K"),
    MOSAIC_TYPE("mosaicType", "Mosaic Group and Type", "K39"),
    MOSAIC_SEGMENT_CODE("mosaicSegmentCode", "Mosaic Segment", "A01_3"),
    FACTOR_1_SCORE("factor1Score", "Factors score", "-24255"),
    FACTOR_1_PERCENTILE("factor1Percentile", "Factors percentile", "13"),
    FACTOR_2_SCORE("factor2Score", "Factors score", "-024255"),
    FACTOR_2_PERCENTILE("factor2Percentile", "Factors percentile", "13"),
    FACTOR_3_SCORE("factor3Score", "Factors score", "-24255"),
    FACTOR_3_PERCENTILE("factor3Percentile", "Factors percentile", "13"),
    FACTOR_4_SCORE("factor4Score", "Factors score", "-24255"),
    FACTOR_4_PERCENTILE("factor4Percentile", "Factors percentile", "13"),
    FACTOR_5_SCORE("factor5Score", "Factors score", "-24255"),
    FACTOR_5_PERCENTILE("factor5Percentile", "Factors percentile", "13"),
    LENGTH_OF_RESIDENCE("lengthOfResidence", "The code representing the Length of Residence band", "2"),
    HEAD_OF_HOUSEHOLD_AGE("headOfHouseholdAge", "The code representing the Age band", "11"),
    PROPENSITY_FOR_CHILDREN_0_TO_10_YEARS("propensityForChildren0To10Years", "Double digit code representing the band", "02"),
    PROPENSITY_FOR_CHILDREN_11_TO_18_YEARS("propensityForChildren11To18Years", "Double digit code representing the band", "02"),
    ADULTS_AT_ADDRESS("adultsAtAddress", "An estimate of the number of adults at the address", "3"),
    HOUSEHOLD_COMPOSITION("householdComposition", "Single digit code representing the Household Composition band", "2"),
    LIFESTAGE("lifestage", "2-digit code representing the Lifestage band", "09"),
    HOUSEHOLD_INCOME("householdIncome", "Single digit code representing the Household Income band", "5"),
    AFFLUENCE("affluence", "Single digit code representing the Affluence band", "3"),
    RISK_INSIGHT("riskInsight", "Code representing the Risk Insight band", "7"),
    CREDIT_DEMAND("creditDemand", "Code representing the Credit Demand band", "7");

    private final String elementName;
    private final String description;
    private final String example;

    Aug(String elementName, String description, String example) {
        this.elementName = elementName;
        this.description = description;
        this.example = example;
    }

    public String getElementName() {
        return elementName;
    }

    public String getDescription() {
        return description;
    }

    public String getExample() {
        return example;
    }

}